### PR TITLE
Chapter1/homework/slowest key press

### PR DESCRIPTION
**Pull request by domiiii11**
This is my godzilla code for today, but  I  have started to rearrange it. :))

### CODE REVIEW

File: src/main/java/edu/SlowestKeySolution.java [hunk 1]
```diff
@@ -1,9 +1,78 @@
...
+            return INVALID;
+        }
+        char[] ch = new char[keysPressed.length()];
+                for (int i = 0; i < keysPressed.length(); i++) {
+                    ch[i] = keysPressed.charAt(i);
+                }
+                System.out.println(ch);
+        int  b = (int) ch[0];
+        char tempChar2 = '0';
+        System.out.println(b);
```
Comment (Jolita-byte) [inline src/main/java/edu/SlowestKeySolution.java]: Task does not require to print out anything.

File: src/main/java/edu/SlowestKeySolution.java [hunk 2]
```diff
@@ -1,9 +1,78 @@
...
+
 public class SlowestKeySolution {
     public static final char INVALID = '-';
 
+
+//    String keysPressed = "asdagrehhg";
+//    int[] releaseTimes = new int[] { 1, 3, 4, 5, 8, 9, 10, 13, 14, 17};
+//        SlowestKeySolution.solve(releaseTimes, keysPressed);
+
     public static char solve(int[] releaseTimes, String keysPressed) {
```
Comment (Jolita-byte) [inline src/main/java/edu/SlowestKeySolution.java:6]: Whole this method can be refactored into smaller ones for easy understanding and better testing

File: src/main/java/edu/SlowestKeySolution.java [hunk 3]
```diff
@@ -1,9 +1,78 @@
...
+        int  b = (int) ch[0];
+        char tempChar2 = '0';
+        System.out.println(b);
+                int[] durations = new int[releaseTimes.length];
+                int tempInt = 0;
+                char tempChar = 'a';
+                durations[0] = releaseTimes[0];
+                System.out.println(Arrays.toString(durations));
+                for (int i = 1; i < releaseTimes.length; i++) {
+                    int releaseTimesDuration = releaseTimes[i] - releaseTimes[i - 1];
```
Comment (Jolita-byte) [inline src/main/java/edu/SlowestKeySolution.java]: There is no need to initiate a variable every loop iteration. The result can be assigned directly to durations [i].

File: src/main/java/edu/Demo.java [hunk 4]
```diff
@@ -1,7 +1,10 @@
 package edu;
 
 public class Demo {
+
     public static void main(String[] args) {
-        System.out.println("Hello!");
+        String keysPressed = "asdagrehhg";
+        int[] releaseTimes = new int[] {1, 3, 4, 5, 8, 9, 10, 13, 14, 17};
+        SlowestKeySolution.solve(releaseTimes, keysPressed);
```
Comment (Jolita-byte) [inline src/main/java/edu/Demo.java]: No need to have testing data and Main class if we have test units

File: src/main/java/edu/SlowestKeySolution.java (hunk 1 above)
Comment (domiiii11) [reply src/main/java/edu/SlowestKeySolution.java]: Yes 💯. Still didn't finish.

File: src/main/java/edu/SlowestKeySolution.java (hunk 2 above)
Comment (domiiii11) [reply src/main/java/edu/SlowestKeySolution.java:6]: My this evening task numb. 1 :))

File: src/main/java/edu/SlowestKeySolution.java (hunk 1 above)
Comment (Jolita-byte) [reply src/main/java/edu/SlowestKeySolution.java]: Cool :) If I understand correctly, in your situation PR was created too early :)

Comment (Almantask) [reply src/main/java/edu/SlowestKeySolution.java]: image
You can create PR early too, just make sure it's marked as draft to indicate work in progress

Comment (domiiii11) [reply src/main/java/edu/SlowestKeySolution.java]: ok ^^

**Comment by domiiii11** [comment]
All tests passed. Would like to move to the next task.

**Pull request closed without merging**